core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -283,6 +283,9 @@ class ReplicaManager(val config: KafkaConfig,
   newGauge("AtMinIsrPartitionCount", () => leaderPartitionsIterator.count(_.isAtMinIsr))
   newGauge("OneAboveMinIsrPartitionCount", () => leaderPartitionsIterator.count(_.isOneAboveMinIsr))
   newGauge("ReassigningPartitions", () => reassigningPartitionsCount)
+  Partition.ISR_STATES_TO_CREATE_METRICS.foreach(c =>
```

Comment:
Would it be helpful? After thinking it for a bit, I feel like I'd not check this metric frequently.

Reply:
This is the kind of metrics that not being frequently used, but once we get into the similar situation that's the best way for us to inspect the internals.  Otherwise we can only rely on heapdumps.

Reply:
Let's say we get the count of each ISR state, and how are we going to use those numbers?

Reply:
In terms of monitoring, we should monitor on the states of `Pending`s.  If it is raise too high (e.g. 30% of leader replicas), then it is problematic and requires manual intervention (e.g. restart)